### ISSUE TITLE
KY040 und IR Control?

### ISSUE THREAD

**Issue by djuscha** [Edzelf/ESP32-Radio#17]
Can one connect KY040 and IR remote control?

**Comment by Edzelf** [comment]
I don't know what a KY040 is.  There is currently no support fo IR remote control.

**Comment by djuscha** [comment]
KY040 is 360 Rotary Encoder Module Brick Sensor Development 
[https://ru.aliexpress.com/item/360-Degree-Rotary-Encoder-Module-Rotary-Code-Module-Encoder-Brick-Sensor-Development-Board-KY-040-Count/32629616836.html?spm=2114.10010208.1000016.1.36Vagi&isOrigTitle=true](url)

**Comment by koskee** [comment]
Although more features is always going to be better, personally I don't think it would necessarily be worth the time invested for the IR remote. The radio already has the web interface, which is quite easily accessible with any phone or tablet - I personally have my phone within arms reach pretty much 24/7 so it's never been a problem. Not to mention, controlling over WiFi doesn't require line of sight, so I can turn mine off after I've already left and gotten into my vehicle outside,  for example. 

Also, IR requires fairly strict timings for proper communication if I recall correctly (unless the task was delegated to an external module of course, but this adds physical bulk to the system). 
The required frequency (40-50khz, don't remember exact number) might not play nice since the mcu already has a decent sized task between filling/transferring data to/from the ringbuf fer and hosting a webserver. 

It's not a bad idea, and if you could turn off the either the webserver or the IR, it should be able to keep up in that case.. It's just that I don't think too many people would be keen on adding another remote to the collection on their coffee table. The technology seems a bit dated Imo. 

That being said, if you have the time and/or skills to implement such a feature, I say go for it. 

OTOH, I had considered the rotary encoder myself, I think that would be beneficial (cut down on necessary buttons for physical control)

**Comment by djuscha** [comment]
Understand :), but encoder would be super :)

**Comment by Edzelf** [comment]
IR interface is implemented now.

**Comment by djuscha** [comment]
:))

**Comment by djuscha** [comment]
And how is it connected? PDF not yet available :(

**Comment by Edzelf** [comment]
Forgot to upload it.  Fixed.

**Comment by djuscha** [comment]
When I write in the config
Ir_40BF = upvolume = 2
Ir_C03F = downvolume = 2
Ir_1F6B = mute

Is in debug log

D: writeprefs ir_8A31 = uppreset = 1
D: writeprefs ir_719A = station = us1.internet-radio.com:8105
D: writeprefs ir_1F6B = mute
D: End of writing preferences
  After the save and restart is the stored is no longer there?
That was yes and if I press buttons does not happen in the log?
what am I doing wrong?

**Comment by Edzelf** [comment]
Try lowercase.

**Comment by djuscha** [comment]
Do you mean that?
ir_40bf = upvolume = 2
ir_c03f = downvolume = 2
ir_1f6b = mute

log
D: writeprefs ir_40bf = upvolume = 2
D: writeprefs ir_c03f = downvolume = 2
D: writeprefs ir_1f6b = mute
D: HTTP input shorter than expected
D: End of writing preferences

Is the same .. after save and restart everything away

**Comment by Edzelf** [comment]
Yes, lower case.  I will check it.  Check ik you can change other prferences, for exmple "volume".

**Comment by GabeHC** [comment]
No where to found any IR related code, Did you not yet committed?

**Comment by Edzelf** [comment]
Uploaded last version.  IR should work now.  Improved MP3 player.

**Comment by djuscha** [comment]
Now everything is wonderful. Everything works! Thank you very much
Now I'm waiting for integration of the encoder :)

**Comment by Edzelf** [comment]
What would you like to with (one?) encoder?

**Comment by djuscha** [comment]
Switch Radiosender, stop / play and volume control.

**Comment by Edzelf** [comment]
I ordered some encoders.

**Comment by djuscha** [comment]
👍 👍 👍

**Comment by djuscha** [comment]
I have that made with Termite simply connected with USB with computer and buttons at the Vernbedinung actuated

 

https://www.compuphase.com/software_termite.htm

 

 

Von: Gabriel Huang [mailto:notifications@github.com] 
Gesendet: Sonntag, 2. Juli 2017 08:38
An: Edzelf/ESP32-Radio <ESP32-Radio@noreply.github.com>
Cc: djuscha <ASDjuscha@t-online.de>; Author <author@noreply.github.com>
Betreff: Re: [Edzelf/ESP32-Radio] KY040 und IR Control? (#17)

 

No where to found any IR related code, Did you not yet committed?

**Comment by Edzelf** [comment]
Serial monitor in Arduino IDE will do as well.

**Comment by Edzelf** [comment]
Rotary encoder is implemented now.  See documentation for functional description.

**Comment by lonelyjack** [comment]
How can i activate ecncoder?

**Comment by Edzelf** [comment]
See the documentation under "Preferences".
I used:
```
ir_pin = 35                                       # GPIO Pin number for IR receiver VS1838B
vs_cs = 5                                         # GPIO Pin number for VS1053 "CS"
vs_dcs = 16                                       # GPIO Pin number for VS1053 "DCS"
vs_dreq = 4                                       # GPIO Pin number for VS1053 "DREQ"
tft_cs = 15                                       # GPIO Pin number for TFT "CS"
tft_dc = 2                                        # GPIO Pin number for TFT "DC"
enc_clk = 25                                      # GPIO Pin number for rotary encoder "CLK"
enc_dt = 26                                       # GPIO Pin number for rotary encoder "DT"
enc_sw = 27                                       # GPIO Pin number for rotary encoder "SW"
sd_cs = 21                                        # GPIO Pin number for SD card "CS"

```

**Comment by djuscha** [comment]
Rotary Encoder Works great! Thanks!!!
